src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -529,16 +537,26 @@ private CreationResultStatus EnterMaintenanceFlow()
                 argsError = !ValidateRemainingParameters();
             }
 
-            if (argsError || IsHelpFlagSpecified)
+            if(IsHelpFlagSpecified)
             {
-                if(commandParseFailureMessage != null)
+                if (commandParseFailureMessage != null)
                 {
                     Reporter.Error.WriteLine(commandParseFailureMessage.Bold().Red());
                 }
 
                 ShowUsageHelp();
                 ShowTemplateHelp(UnambiguousTemplateToUse.Info);
-                return argsError ? CreationResultStatus.InvalidParamValues : CreationResultStatus.Success;
+                return CreationResultStatus.Success;
```

Comment:
@phenning I wasn't quite sure what to do with that state - there's an arg error, but the user asked for help, so help was successfully shown but they were also notified of an error. You'd vote for putting it back to return error if there's an error even though help was specified? (I'm down with that, just looking for someone to overcome my indecision)